packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -17,7 +17,6 @@ interface AstroPluginOptions {
 }
 
 const VIRTUAL_MODULE_ID_PREFIX = 'astro:markdown';
-const VIRTUAL_MODULE_ID = '\0' + VIRTUAL_MODULE_ID_PREFIX;
```

Comment:
This `\0` prefix breaks module graph resolution for some reason. Since it appends `/@id/` to each entry, I assume this character messes with the path construction. Removing with the assumption others won't slap the `astro:markdown` prefix on us [as warned about here](https://vitejs.dev/guide/api-plugin.html#virtual-modules-convention)!

Reply:
can you leave a comment here that `\0` breaks things, so that a future reader doesn't think to add it back?

Reply:
switched back to special `?mdImport` flag bc of module resolution bugs. Removes the `VIRTUAL_MODULE_ID`!